models/Document/Editable.php

```diff
@@ -711,4 +713,20 @@ public function setEditableDefinitionCollector(?EditmodeEditableDefinitionCollec
 
         return $this;
     }
+
+    protected function getUnserializedData(mixed $data): ?array
+    {
+        if (is_array($data) || is_null($data)) {
```

Comment:
We should avoid if elseif else constructs since it makes the code kind of unreadable and more complex than it needs to be.

The whole function could be rewritten alongs the lines of 

```php
    if(!is_array($data) && !is_null($data) && !is_string($data)) {
            throw new InvalidArgumentException('Data must be a string, an array or null.');
        }

        if (is_array($data) || is_null($data)) {
            return $data;
        }

        
        $unserializedData = Serialize::unserialize($data);
        if (!is_array($unserializedData) && !is_null($unserializedData)) {
            throw new InvalidArgumentException('Unserialized data must be an array or null.');
        }

        return $unserializedData;
```

WDTY?

Reply:
@mattamon You are right. This wasn't possible before :)
See https://github.com/pimcore/pimcore/pull/16773/commits/ad70c75c75d0c079d46c69c5f2f8eefd8050063b